<launch>

  <arg name="bagfiles" default="/home/plnegre/bagfiles/images170927_1234_2017-09-27-10-43-51_17.bag"/>

  <arg name="camera" default="stereo_forward"/>
  <arg name="image" default="image_raw"/>

  <param name="use_sim_time" value="true"/>

  <node pkg="rosbag" type="play" name="player" args="--clock -d 5 -l $(arg bagfiles)"/>

  <group ns="$(arg camera)">
    <node pkg="stereo_image_proc" type="stereo_image_proc" name="stereo_image_proc"/>
  </group>

  <group ns="/vig/$(arg camera)">
    <node pkg="stereo_image_proc" type="stereo_image_proc" name="stereo_image_proc"/>
  </group>

  <node pkg="image_preprocessing" type="vignetting_remover" name="vignetting_remover_l" output="screen">
    <param name="mean_filename" value="$(find image_preprocessing)/data/vignetting_mean_left.png"/>
    <param name="std_filename" value="$(find image_preprocessing)/data/vignetting_std_left.png"/>
    <param name="output_namespace" value="/vig/$(arg camera)/left"/>
    <remap from="image" to="$(arg camera)/left/$(arg image)"/>
  </node>
  <node pkg="image_preprocessing" type="vignetting_remover" name="vignetting_remover_r" output="screen">
    <param name="mean_filename" value="$(find image_preprocessing)/data/vignetting_mean_right.png"/>
    <param name="std_filename" value="$(find image_preprocessing)/data/vignetting_std_right.png"/>
    <param name="output_namespace" value="/vig/$(arg camera)/right"/>
    <remap from="image" to="$(arg camera)/right/$(arg image)"/>
  </node>

  <node pkg="image_preprocessing" type="color_correction_image" name="color_correction_image" output="screen">
    <remap from="/color_correction_image/input" to="/vig/$(arg camera)/left/image_rect_color"/>
    <remap from="/color_correction_image/output" to="/vig/$(arg camera)/left/image_rect_color_corrected"/>
  </node>

</launch>
